module/VuFindApi/src/VuFindApi/Controller/AdminApiController.php

```diff
@@ -0,0 +1,173 @@
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA    02111-1307    USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+namespace VuFindApi\Controller;
+
+use Laminas\ServiceManager\ServiceLocatorInterface;
+use VuFind\Cache\Manager as CacheManager;
+
+/**
+ * Admin Api Controller
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class AdminApiController extends \VuFind\Controller\AbstractBase
+    implements ApiInterface
+{
+    use ApiTrait;
+
+    /**
+     * Cache manager
+     *
+     * @var CacheManager
+     */
+    protected $cacheManager;
+
+    /**
+     * Constructor
+     *
+     * @param ServiceLocatorInterface $sm Service locator
+     * @param CacheManager            $cm Cache manager
+     */
+    public function __construct(ServiceLocatorInterface $sm, CacheManager $cm)
+    {
+        $this->serviceLocator = $sm;
+        $this->cacheManager = $cm;
+    }
+
+    /**
+     * Permission required for the clear cache endpoint
+     *
+     * @var string
+     */
+    protected $cacheAccessPermission = 'access.api.admin.cache';
+
+    /**
+     * Caches that are not cleared by the clearCache command by default
+     *
+     * @var array
+     */
+    protected $defaultIgnoredCaches = ['cover'];
+
+    /**
+     * Clear the cache
+     *
+     * @return \Laminas\Http\Response
+     */
+    public function clearCacheAction()
+    {
+        $this->disableSessionWrites();
+        $this->determineOutputMode();
+
+        if ($result = $this->isAccessDenied($this->cacheAccessPermission)) {
+            return $result;
+        }
+
+        try {
+            $cacheList = $this->getRequest()->getQuery()->get('id')
+                ?: $this->getDefaultCachesToClear();
+            foreach ((array)$cacheList as $id) {
+                $cache = $this->cacheManager->getCache($id);
+                if (!($cache instanceof \Laminas\Cache\Storage\FlushableInterface)) {
+                    throw new \Exception("Cache $id is not flushable");
```

Comment:
Do we want a single unflushable cache to kill the whole process? Couldn't this cause some caches to get flushed and others to get skipped, if the problem happens in the middle of the sequence? Would it be better to collect an array of cache-specific messages and then report them all at the end of the process instead of letting the first problem terminate the process? I'm thinking we could have a structure like:

<pre>
{
    'errorCount': 1,
    'details': [
        { 'name': 'cache1', 'error': false, 'message': 'ok' },
        { 'name': 'cache2', 'error': true, 'message': 'Cache cache2 is not flushable' },
        { 'name': 'cache3', 'error': false, 'message': 'ok' },
    ]
}
</pre>

Maybe this is overkill, but it might allow finer-grained control. If you don't think it's worth the effort now, we could always open it as a possible future enhancement ticket.

Reply:
Now that the caches are filtered, I think this shouldn't be necessary. If you feel it'd still be needed, I can add it, but I wouldn't expect issues normally.

Reply:
We can always add it in the future if we need it; as you say, probably not a requirement at this point.